docs/specification.md

```diff
@@ -0,0 +1,1199 @@
+SPDX-License-Identifier: MIT
+-->
+
+**Overview**
+------------
+
+These specifications were assembled with the following references:
+
+- [*CENTRE Fiat Token design*](https://github.com/centrehq/centre-tokens/blob/78d964a1a8d481ffd8152772d7a66e47df54b3db/doc/tokendesign.md) which is served as a reference specification to tezos stablecoin smart contract.
+
+- Tezos Token Standard:
+  [*FA2*](https://gitlab.com/tzip/tzip/-/blob/76d5f3791bfbfe3c9bf95ad5ec5fc6cbeeca2d0e/proposals/tzip-12/tzip-12.md) and
+  [*ManagedLedger*](https://gitlab.com/tzip/tzip/blob/ae2f1e7ebb3454d811a2bea3cd0698b0e64ccea5/proposals/tzip-7/ManagedLedger.md)
+
+- [*Michelson Contract Interfaces and Conventions*](https://gitlab.com/tzip/tzip/blob/ae2f1e7ebb3454d811a2bea3cd0698b0e64ccea5/proposals/tzip-4/tzip-4.md) TZIP which defines `view` and `void` type synonyms
+
+**General Requirements**
+------------------------
+
+- The token(s) must be and FA2 compatible as to facilitate listing on
+  exchanges and interaction with services which support FA2
+
+**Ledger**
+==========
+
+Every address that is stored in ledger is associated with its
+current balance, token id and current minting allowance.
+
+**Roles**
+=========
+
+The token supports the following list of user roles as is described in
+CENTRE Fiat Token specification and the taxonomy of permission policies that are described in FA2 token specification.
+
+<!-- Here're the roles that are present in current version of stablecoin token.  -->
+
+Below we define some roles for stablecoin contract token.
+
+**owner**
+
+- Can re-assign any role of the token.
+
+**masterMinter**
+
+- Can add and remove minters.
+
+- Can change minting allowance for any minter if that is present in
+  its list of minters.
+
+- Stores the list of current minters within itself, rather
+  than exposing the map of master minter and its minters globally
+  for the whole token.
+
+**minter**
+
+- Can create and destroy coins (that are allowed by their current
+  minting allowance).
+
+- Minting allowance is stored locally within the address
+  allowing for multiple minters creating and destroying tokens.
+
+**pauser**
+
+- Can pause transferring, burning and minting operations.
+  During the pause, these operations cannot be performed
+  and fail with an error if the user decides to try them.
+
+**blacklister (TBD)**
+
+- Can blacklist a particular address preventing it from
+  transferring, minting, burning and receiving tokens by
+  removing them from whitelist.
+
+**Token Functions**
+===================
+
+**Standard Token Functions**
+----------------------------
+
+Functions for the stablecoin token implementation which are common to the [*FA2 Tezos
+Token Standard*](https://gitlab.com/tzip/tzip/-/blob/76d5f3791bfbfe3c9bf95ad5ec5fc6cbeeca2d0e/proposals/tzip-12/tzip-12.md). The
+token implementation in general follows an [*event loop*](https://en.wikipedia.org/wiki/Event_loop) pattern meaning that every parameter
```

Comment:
AFAIU it only applies to getters that indeed take a callback contract. When you say that the token implementation in general follows an event loop, it makes one think that it applies to nearly everything that is going on in the contract.

Anyway, what's the purpose of this sentence? Passing a callback contract is a standard approach and basically the only way to get data from another contract in Tezos.